sdk-rs/src/websocket_program_account_subscriber.rs

```diff
@@ -0,0 +1,184 @@
+use std::pin::Pin;
+use std::sync::Arc;
+use anchor_lang::AccountDeserialize;
+use base64::{Engine as _, engine::general_purpose};
+use futures_util::Future;
+use solana_account_decoder::{UiAccountEncoding, UiAccountData};
+use solana_client::rpc_filter::RpcFilterType;
+use solana_sdk::commitment_config::CommitmentConfig;
+use solana_client::pubsub_client::PubsubClient;
+use solana_client::rpc_config::{RpcProgramAccountsConfig, RpcAccountInfoConfig};
+use crate::types::{DataAndSlot, SdkResult};
+use log::{error, warn};
+use tokio::task::LocalSet;
+
+pub struct WebsocketProgramAccountOptions {
+    filters: Vec<RpcFilterType>,
+    commitment: CommitmentConfig,
+    encoding: UiAccountEncoding,
+}
+
+pub struct WebsocketProgramAccountSubscriber<T>
+where 
+    T: AccountDeserialize + core::fmt::Debug,
+{
+    _subscription_name: String,
+    url: String,
+    options: WebsocketProgramAccountOptions,
+    on_update: Option<Arc<dyn Fn(String, DataAndSlot<T>) -> Pin<Box<dyn Future<Output = ()>>>>>,
+    _resub_timeout_ms: Option<u64>,
+    subscribed: bool,
+}
+
+impl<T> WebsocketProgramAccountSubscriber<T>
+where 
+    T: AccountDeserialize + core::fmt::Debug,
+{
+    pub fn new(
+        subscription_name: String,
+        url: String,
+        options: WebsocketProgramAccountOptions,
+        on_update: Option<Arc<dyn Fn(String, DataAndSlot<T>) -> Pin<Box<dyn Future<Output = ()>>>>>,
```

Comment:
alias this type  `dyn Fn(String, DataAndSlot<T>) -> Pin<Box<dyn Future<Output = ()>>` e.g. `type OnUpdateFn<T>`, its more readable and easier for consumers to use it